pootle/core/management/subcommands.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+import sys
+from collections import OrderedDict
+
+from django.core.management import (
+    BaseCommand, CommandError, CommandParser, handle_default_options)
+from django.core.management.base import SystemCheckError
+from django.db import connections
+from django.utils.functional import cached_property
+
+from pootle.core.delegate import subcommands
+
+
+class SubcommandsError(CommandError):
+
+    pass
+
+
+class SubcommandsParser(CommandParser):
+
+    def parse_args(self, args):
+        try:
+            # first find if we have a known subcommand
+            known, __ = self.parse_known_args(args)
+        except SubcommandsError:
+            known = None
+        is_subcommand = (
+            known
+            and hasattr(known, "subcommand")
+            and known.subcommand in self.cmd.subcommands)
+        if is_subcommand:
+            # if there is a subcommand then move it in argv to the front of other
+            # params and defer to the the subcommand
+            args = list(args)
+            del args[args.index(known.subcommand)]
```

Comment:
I am now thinking that this won't remove the subcommand from the previous `args` object which might be still used in other places. So forget about my proposal.